src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -363,9 +363,20 @@ export class NotebookModel extends Disposable implements INotebookModel {
 						}
 					});
 				}
+				// Get Notebook URI Params & adjust to string
+				let notebookUriParams: string = this.notebookUri?.query;
```

Comment:
Why the ? - it looks like the URI is never undefined?

Reply:
If it can be undefined then you should be updating the property getter to reflect that - right now it's just returning a URI. Make sure you keep an eye out for that kind of stuff here when making changes - we don't currently strictly enforce this in a lot of places but it doesn't mean we shouldn't keep the typings accurate when we have the chance.

But in this case I was specifically talking about the URI being undefined, NOT the query parameters (which I agree can be empty/undefined and should be handled as such). A notebook should always have a URI associated with it and indeed we appear to do a check for this in the constructor

```
if (!_notebookOptions || !_notebookOptions.notebookUri || !_notebookOptions.notebookManagers) {
	throw new Error('path or notebook service not defined');
}
```

so doing this should be fine here : 

`const notebookUriParams: string = this.notebookUri.query;`